### PR TITLE
Update shim bin-placing to use binplace.targets

### PR DESCRIPTION
**Pull request by weshaggard**
Also disable apicompat checks for uap* when in BuildAllConfigurations
mode because it doesn't have a RuntimePath correctly setup for those
configurations.

cc @joperezr @ericstj

### CODE REVIEW

File: src/shims/dir.props [hunk 1]
```diff
@@ -25,6 +26,12 @@
   </ItemGroup>
 
   <PropertyGroup>
-  	<GenFacadesOutputPath>$(BaseIntermediateOutputPath)/shims/facades/</GenFacadesOutputPath>
+    <IsNETCoreApp>true</IsNETCoreApp>
```
Comment (joperezr) [inline src/shims/dir.props]: Do they have to go in the Ref of both packages as well? I would suppose they would only be used at runtime, in which case we can set `IsNETCoreAppRef` and `IsUAPRef` to false.

File: src/shims/shims.proj [hunk 2]
```diff
@@ -43,31 +42,23 @@
...
     <Exec Command="$(GenFacadesCmd) -contracts:&quot;@(NETStandardContracts)&quot; @&quot;$(GenFacadesResponseFile)&quot;" WorkingDirectory="$(ToolRuntimePath)" />
 
-    <!-- Copy the facades to the package ref and lib folders to be included in the packages -->
-    <!-- TODO: replace with BinPlacing targets -->
-    <ItemGroup>
-      <PackageOutputPaths Include="$(BinDir)pkg/$(TargetGroup)/ref" Condition="'$(_bc_TargetGroup)'!='uapaot'"/>
-      <PackageOutputPaths Include="$(BinDir)pkg/uap/ref" Condition="'$(_bc_TargetGroup)'=='uapaot'"/>
-      <PackageOutputPaths Include="$(BinDir)pkg/$(TargetGroup)/lib" />
-      <PackageOutputPaths Include="$(RefPath)" />
-      <PackageOutputPaths Include="$(RuntimePath)" />
```
Comment (joperezr) [inline src/shims/shims.proj]: Will they still get deployed into the RuntimePath? From what I can tell, it looks like they will only get placed into the `bin\pkg` dirs now.

File: src/shims/dir.props (hunk 1 above)
Comment (weshaggard) [inline src/shims/dir.props]: They should be in the ref as well for compile-time type alignment.

File: src/shims/shims.proj (hunk 2 above)
Comment (weshaggard) [inline src/shims/shims.proj]: See https://github.com/dotnet/corefx/blob/master/binplace.targets#L31 they will till be put into the RuntimePath.

**Comment by ericstj** [comment]
Cool, LGTM.

**Pull request merged**